backend/de.metas.async/src/main/java/de/metas/async/api/impl/QueueProcessorDAO.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * #%L
+ * de.metas.async
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.async.api.impl;
+
+import de.metas.async.api.IQueueDAO;
+import de.metas.async.model.I_C_Queue_PackageProcessor;
+import de.metas.async.model.I_C_Queue_Processor;
+import de.metas.async.model.I_C_Queue_Processor_Assign;
+import de.metas.async.processor.QueueProcessorId;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.compiere.Adempiere;
+import org.compiere.SpringContextHolder;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+@Repository
+public class QueueProcessorDAO
+{
+	public static QueueProcessorDAO getInstance()
+	{
+		if (Adempiere.isUnitTestMode())
+		{
+			return new QueueProcessorDAO();
+		}
+		else
+		{
+			return SpringContextHolder.instance.getBean(QueueProcessorDAO.class);
+		}
+	}
+
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final IQueueDAO queueDAO = Services.get(IQueueDAO.class);
+
+	@NonNull
+	public Optional<QueueProcessorId> getProcessorForPackage(@NonNull final I_C_Queue_PackageProcessor packageProcessor)
+	{
+		final Optional<QueueProcessorId> queueProcessorId = queryBL.createQueryBuilder(I_C_Queue_Processor_Assign.class)
```

Comment:
Why not caching this thing...
e.g. loud thinking:
* have a `QueueProcessorDescriptor` datastructure which contains the queue processor ID and all the assigned package processors
* have a `QueueProcessorDescriptorIndex` data structure which contains ALL QueueProcessorDescriptor and offers nice methods to get a queue processor ID by package processor ID or what ever is needed
* load once all QueueProcessorDescriptorIndex and cache it (have a cache reset when depending tables are changing). 
....

Reply:
will be addresses in follow-up